test/e2e/apimachinery/flowcontrol.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apimachinery
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/onsi/ginkgo"
+	"k8s.io/apiserver/pkg/server/filters"
+	"k8s.io/client-go/rest"
+
+	flowcontrolv1alpha1 "k8s.io/api/flowcontrol/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+var _ = SIGDescribe("APIPriorityAndFairness", func() {
+	f := framework.NewDefaultFramework("flowschemas")
+
+	framework.ConformanceIt("should ensure that requests can be classified by testing flow-schemas/priority-levels", func() {
+		testingFlowSchemaName := "e2e-testing-flowschema"
+		testingPriorityLevelName := "e2e-testing-prioritylevel"
+		matchingUsername := "noxu"
+
+		ginkgo.By("creating a testing prioritylevel")
+		createdPriorityLevel, err := f.ClientSet.FlowcontrolV1alpha1().PriorityLevelConfigurations().Create(
+			context.TODO(),
+			&flowcontrolv1alpha1.PriorityLevelConfiguration{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: testingPriorityLevelName,
+				},
+				Spec: flowcontrolv1alpha1.PriorityLevelConfigurationSpec{
+					Type: flowcontrolv1alpha1.PriorityLevelEnablementLimited,
+					Limited: &flowcontrolv1alpha1.LimitedPriorityLevelConfiguration{
+						AssuredConcurrencyShares: 1, // will have at minimum 1 concurrency share
+						LimitResponse: flowcontrolv1alpha1.LimitResponse{
+							Type: flowcontrolv1alpha1.LimitResponseTypeReject,
+						},
+					},
+				},
+			},
+			metav1.CreateOptions{})
+		framework.ExpectNoError(err, "failed to create a testing priority-level")
+
+		ginkgo.By("creating a testing flowschema")
+		createdFlowSchema, err := f.ClientSet.FlowcontrolV1alpha1().FlowSchemas().Create(
+			context.TODO(),
+			&flowcontrolv1alpha1.FlowSchema{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: testingFlowSchemaName,
+				},
+				Spec: flowcontrolv1alpha1.FlowSchemaSpec{
+					MatchingPrecedence: 1000, // a rather higher precedence to ensure it make effect
```

Comment:
is this so that default flow schemas don't interfere?

Reply:
exactly